### ISSUE TITLE
PHPUnit 6 Support

### ISSUE THREAD

**Issue by ntzm** [humbug/humbug#204]
`--tap` has been removed as of PHPUnit 6, any plans to support the new version?

Thanks!

**Comment by alexhoma** [comment]
I've got this:
```
Tests must be in a fully passing state before Humbug is run.
Incomplete, skipped or risky tests are allowed.
The testing framework reported an exit code of 1.
Stdout:
   > PHPUnit 6.0.10 by Sebastian Bergmann and contributors.
   > 
   > unrecognized option --tap
   > 
```

**Comment by ntzm** [comment]
@alexhoma It's not compatible at the moment, use PHPUnit 5.7 if you want Humbug support

**Comment by alexhoma** [comment]
Thanks @ntzm !

**Comment by castarco** [comment]
Is there much difference between the TAP output and the JSON or XML outputs? Maybe I can convince my boss to dedicate some time to implement a new parser.

**Comment by theofidry** [comment]
@castarco I'm not very familiar with it, but it doesn't look too complicated to do a switch.

So if you can find time for a PR, that would be much appreciated :) I will fork the repo soon to take care of some issues until padraic find some time to tidy up things.

**Comment by padraic** [comment]
See #237 which switches output format to Teamcity.

**Comment by ntzm** [comment]
Legend mate, nice one

**Comment by omissis** [comment]
@padraic I was diving into the code and yesterday (unaware of #237) I started tinkering with swapping tap with junit xml (I do also have a phpunit6 branch with deps and BCs out of the way btw). I guess I should stop if you're working on teamcity right? :)

**Comment by theofidry** [comment]
Actually I think having a support for JUnit would be good as well as it's quite a standard format. But admittedly it's not a high priority

**Comment by omissis** [comment]
@theofidry I was trying to swap tap with junit so to be able to move to phpunit6, but if @padraic is doing the same with teamcity, it's probably not worth duplicating the effort.

**Comment by theofidry** [comment]
if we can only support one yeah it doesn't make sense, but I think it makes sense to have multiple adapters. So it's not duplicating the effort, but just working on another feature :) As I said though it's not a high priority so obviously don't feel obliged to

**Comment by theofidry** [comment]
if we can only support one yeah it doesn't make sense, but I think it makes sense to have multiple adapters. So it's not duplicating the effort, but just working on another feature :) As I said though it's not a high priority so obviously don't feel obliged to

**Comment by dmecke** [comment]
Is there anything I could do to help?

**Comment by theofidry** [comment]
@dmecke now that the tap support has been merged we need to check what is missing to offer the compatibility :)

**Comment by omissis** [comment]
@theofidry @dmecke I already have a branch with what needed to be done, I just have to rebase and recheck, then I can send the PR in. I'll see if I can do it this evening.

**Comment by padraic** [comment]
Teamcity support, not TAP ;). I was checking TAP support in PHPUnit, and it was all over the place from PHPUnit 4 - odd spare text in output, exceptions, inconsistent results. This will have solved a couple of the "passes when I run it without Humbug" initial run problems.

On another note, this issue can be closed for original description - if there are related PHPUnit issues (dependency isolation, version updates, compatibility updates), we can open new issues. I'll leave this open for a while.

**Comment by omissis** [comment]
quick update: I have rebased and I have some tests failing. I am on it though, I hope I'll be able send a PR in soon ;)

**Comment by omissis** [comment]
another quick update: I am still working on it in my spare time. I've recently sent a pr for updating phpunit-accelerator (one of dependencies) to phpunit6, and I am wrapping my head around those failing tests.

**Comment by exussum12** [comment]
Any further update on this @omissis ?

**Comment by omissis** [comment]
hey @exussum12: unfortunately not, I didn't have time to work on it anymore since then. Still on my roadmap, as soon as I'll get out of before-vacations crunch :)

**Comment by TrogWarZ** [comment]
Any info about this?

**Comment by theofidry** [comment]
#237 has been merged but I didn't find time to check out what has been missing. But maybe you can try it out :)

**Comment by omissis** [comment]
@TrogWarZ I have been looking at the matter in very sparse bursts, but between having little familiarity with the code and the lack of time, I didn't progress as I wished. I don't think it will take too much, but yeah, time. I have been working on this branch in case you are willing to take a look: https://github.com/omissis/humbug/tree/feature/phpunit6-support. Any help is welcome :)

**Comment by TrogWarZ** [comment]
Thanks @theofidry and @omissis , i'll try to help with that on the next week. But i need to know how to not trigger CI errors (-:

**Comment by exussum12** [comment]
Looks like support had been merged ? 

https://github.com/humbug/humbug/blob/master/composer.json

**Comment by theofidry** [comment]
Yeah but still broken. We need to patch Mockery to be able to finish that move :/

**Comment by robertbasic** [comment]
@theofidry what patch is required in Mockery? Mockery v1 was tagged yesterday, btw: https://github.com/mockery/mockery/releases/tag/1.0

**Comment by theofidry** [comment]
Oh indeed, well I need to check out if there is anything else missing then :)

**Comment by theofidry** [comment]
Closed via #270. Thanks everyone for getting involved. I hope to tag the stable release (the RC release has just been tagged now) in a week or two and then we can get started with Humbug 2.0 :)
